src/PoolManager.sol

```diff
@@ -527,11 +523,11 @@ contract PoolManager is Auth {
     }
 
     function getLiquidityPool(uint64 poolId, bytes16 trancheId, uint128 currencyId) public view returns (address) {
-        return pools[poolId].tranches[trancheId].liquidityPools[currencyIdToAddress[currencyId]];
+        return TrancheTokenLike(pools[poolId].tranches[trancheId].token).vault(currencyIdToAddress[currencyId]);
```

Comment:
Bit more convoluted than before, but it seems like a necessary change to conform to ERC7575?

Reply:
Yes exactly, this is part of ERC7575. I do think this is beneficial for integrations, they can easily find a vault (liquidity pool) for any asset, from the tranche token contract.